src/core/Expander/Expander/Expander.tsx

```diff
@@ -0,0 +1,226 @@
+import React, { Component, ReactNode } from 'react';
+import { default as styled } from 'styled-components';
+import classnames from 'classnames';
+import { withSuomifiDefaultProps } from '../../theme/utils';
+import { AutoId } from '../../../utils/AutoId';
+import { TokensProp, InternalTokensProp } from '../../theme';
+import { HtmlDiv } from '../../../reset';
+import { baseStyles } from './Expander.baseStyles';
+import {
+  ExpanderGroupConsumer,
+  ExpanderGroupProviderState,
+} from '../ExpanderGroup/ExpanderGroup';
+
+const baseClassName = 'fi-expander';
+const openClassName = `${baseClassName}--open`;
+
+export interface ExpanderProviderState {
+  /** Callback for communicating ExpanderTitle button event to Expander  */
+  onToggleExpander: () => void;
+  /** Open state for expander */
+  open: boolean;
+  /** Id for expander button */
+  titleId: string | undefined;
+  /** Id for expander content */
+  contentId: string | undefined;
+}
+
+const defaultProviderValue: ExpanderProviderState = {
+  onToggleExpander: () => null,
+  open: false,
+  titleId: undefined,
+  contentId: undefined,
+};
+
+const {
+  Provider: ExpanderProvider,
+  Consumer: ExpanderConsumer,
+} = React.createContext(defaultProviderValue);
+
+interface InternalExpanderProps {
+  /**
+   * Children, extend type ExpanderTitleBaseProps or ExpanderContentBaseProps
+   * ExpanderProviderState context is used to communicate between title, content and expander
+   */
+  children: ReactNode;
+  /** Custom classname to extend or customize */
+  className?: string;
+  /**
+   * Id for expander, must be unique. Duplicate id's break ExpanderGroup functionality.
+   * Autogenerated if not provided
+   */
+  id?: string;
+  /** Default status of expander open
+   * @default false
+   */
+  defaultOpen?: boolean;
+  /** Controlled open property */
+  open?: boolean;
+  /** Event handler to execute when clicked */
+  onOpenChange?: (open: boolean) => void;
+  consumer?: ExpanderGroupProviderState;
+}
+
+export interface ExpanderProps extends InternalExpanderProps, TokensProp {}
+
+export interface ExpanderTitleBaseProps {
+  /** Custom classname to extend or customize */
+  className?: string;
+  /** Expander consumer for open state and toggle open callback */
+  consumer: ExpanderProviderState;
+}
```

Comment:
Should this interface be here instead
`suomifi-ui-components/src/core/Expander/ExpanderTitle/ExpanderTitle.tsx` ?

Reply:
In my opinion no. It is the constraint for the expander interface and is required to be implemented by all title components. E.g. ExpanderTitleButton and future ExpanderTitle need to implement / extend this interface (in addition to some implementation details) to work.

Reply:
Then it is good to have it here. 

By reviewing the PR, it is hard to know all the future use cases and therefore hard to see reason why it is **now** here. As we try to prevent making features and such that are _maybe_ used in the future.

But as discussed, that you will soon in separate PR create those other possible title components, then this is much clearer.

Reply:
Yes, this is exported now, but used only in one place. This will change shortly after this PR gets merged. Wasn't sure about the pattern my self, but could not come up with anything better since it really is a constraint for the children of Expander.